lib/migration.js

```diff
@@ -0,0 +1,558 @@
+var async = require('async');
+
+module.exports = mixinMigration;
+
+function mixinMigration(MsSQL) {
+  MsSQL.prototype.showFields = function(model, cb) {
+    var sql = 'select [COLUMN_NAME] as [Field], '
+      + ' [IS_NULLABLE] as [Null], [DATA_TYPE] as [Type],'
+      + ' [CHARACTER_MAXIMUM_LENGTH] as [Length],'
+      + ' [NUMERIC_PRECISION] as [Precision], NUMERIC_SCALE as [Scale]'
+      + ' from INFORMATION_SCHEMA.COLUMNS'
+      + ' where [TABLE_SCHEMA] = \'' + this.schema(model) + '\''
+      + ' and [TABLE_NAME] = \'' + this.table(model) + '\''
+      + ' order by [ORDINAL_POSITION]'
+    this.execute(sql, function(err, fields) {
+      if (err) {
+        return cb && cb(err);
+      } else {
+        if (Array.isArray(fields)) {
+          fields.forEach(function(f) {
+            if (f.Length) {
+              f.Type = f.Type + '(' + f.Length + ')';
+            } else if (f.Precision) {
+              f.Type = f.Type + '(' + f.Precision, +',' + f.Scale + ')';
+            }
+          });
+        }
+        cb && cb(err, fields);
+      }
+    });
+  };
+
+  MsSQL.prototype.showIndexes = function(model, cb) {
+    // TODO: [rfeng] Implement SHOW INDEXES
```

Comment:
The index support is not implemented in MsSQL yet. It could be a task of the migration/discovery refactor/cleanup. Let's focus on the CRUD impl for now.